### ISSUE TITLE
Wolf Pet with Wolf Alarm On (Ticking Player Crash)

### ISSUE THREAD

**Issue by Coraline77447** [capnkirok/Inventory-Pets#456]
Minecraft: 1.16.5
Forge: 36.1.23
Inventory Pets: 1.16.5-2.0.14

When I found the wolf pet everything was fine till I turned on the wolf alarm and it kept alerting me that there's an enderman nearby and when I was sort of fighting a zombie (the zombie was at a cave entrance but he couldn't reach me tho) and the game suddenly crashed. 
When I came back in I had to turn off the wolf alarm so that the game wouldn't crash again.
I retrieved the wolf back into the wolf pet and put the item in my special diamond chest (from Iron Chests mod btw) that I keep right next to my bed and several times the wolf from the wolf pet wouldn't despawn back into the item and had to get rid of the wolf pet since it's glitchy for me and the wolf wasn't helping me much 
(especially against an Apotheosis boss which is either a normal zombie, husk, or stray but all beefed up and is equipped with mad enchanted armor and weapons).

Anyways here's the log and crash report:

Log: [latest.log](https://github.com/capnkirok/Inventory-Pets/files/6584562/latest.log
Crash Report: https://pastebin.com/s1zx6V4k

**Comment by capnkirok** [comment]
Thank you for the detailed report and the logs. I really appreciate the follow up. So, somehow the pet got disconnected from its owner and caused this crash (and the behavior you saw). I can put in a workaround so the crash doesn't happen. Will report back once it's fixed.

**Comment by Coraline77447** [comment]
You're welcome, btw one thing I'd forgot to mention was when I still had the wolf pet which was in my diamond chest that there was no wolf with a specialized name that was supposed to be in the item (I have natural wolfs that had spawned and I've tamed them but they're outside my house).
But for some odd reason that one of my double front doors of my house was opening and closing since I have a pressure plate on the inside while I was outside the house which was caused by an invisible entity which was probably the specially named wolf but invisible somehow. 
Every time I would teleport back to my house from wherever I'm at there was the wolf from the wolf pet which the item itself was in a storage container (my diamond chest).

Also I would like to thank you for making this mod since the Heart Pet has saved my hide several times (especially in boss battles against the Apotheosis bosses even tho those guys are like what I've described in my last comment, also the sand giant from Advent of Ascension aka Nevermine (currently hunting the sand giant for hunter xp and regular xp) and both the iron golem pet for providing me with absorption IV or V and the Pixin XP Pet for extra xp.

One last thing, I hope that the anvil pet would get fixed since I've seen some reports on here that the anvil pet ain't working correctly or something and like what someone reported that the feed bag doesn't always work.
It's annoying to go into the feed bag's inventory for my pets to eat even tho sometimes I don't want the blaze pet to eat since I don't always want fire damage and strength effect from it for the regular mobs since I want to use him for boss battles for the extra boost in attack damage and fire damage as the assistant in the boss battles or against tougher enemies than the regular mobs.

So what I'm basically saying is that I wish there was a switch to turn on/off the blaze pet's strength effect so that I can just switch on the blaze pet's strength effect with a shift-right click right before a boss battle or a battle against tougher enemies rather than always active for the regular mobs that I can kill with a powerful weapon from Advent of Ascension (as of a tier 2 weapon to a final tier weapon which may be tier 4 or 5), any powerful affixs from Apotheosis on a sword or even a tinkers' weapon that I could make in Tinkers' Construct. 
Why I'm suggesting this is cause I'm tryin to use up all of my special iron equipment that Apotheosis put on them (affix items) then progress to the void stuff such as the void armor from A.O.A, etc.

P.S. sorry for this comment being an "essay" but I guess I'm one of those types of people who can type really, really, really long messages or something....

**Comment by capnkirok** [comment]
Ah! You just gave me a major clue with the additional wolves outside your house. That is the source of the crash. In this part of the code, it looks to see if there's a nearby tamed wolf, assuming this to be the one spawned from your pet, and tries to despawn it. If there's a conflict around which would be the correct pet to stow (or subsequent behavior), it would cause a crash.

And thanks for the essay. I am working on the other items listed here and may have a solution (in the future) that would allow you to modify the Blaze effect.

**Comment by capnkirok** [comment]
Ok, this is now fixed in development. Look for the updated release tomorrow.

**Comment by capnkirok** [comment]
And that includes the Feed Bag issues :)

**Comment by Coraline77447** [comment]
Sorry for not answering, I don't have any way to get notified from GitHub.
But thanks for fixing the issues.
I should've known that having naturally tamed wolves could've caused a crash which explains why the specially named wolf (named "Dog" at the time) wasn't getting dispatched back into the wolf pet.

With the wolf alarm was probably suppose to come from the special wolf from the wolf pet (which will only come from the special wolf or the wolf pet item itself now) but since I have naturally tamed wolves at my house in my world called Coraline World 2: Nevermine that the mod's code gotten confused as to which wolf was doing the warning calls to the player.

(btw I'm currently playing on Coraline World 3: Chaos Awakens which Chaos Awakens is the official port of Orespawn but it's still in beta and is possibly missing ALOT of content from Orespawn still, so I'm not playing on the world that I've discovered the issue.)

I've downloaded the latest update btw.
